### PR TITLE
[sentry-native] Fix  windows usage

### PR DESCRIPTION
**Pull request by JonLiu1993**
Hi!
I have two issues regarding the using of the sentry-native port.
1. While trying to generate the cmake project build configuration while using `sentry-native` as vcpkg dependency, i'm getting this error:
```
1> [CMake] CMake Error at out/build/Windows-x86-Debug/vcpkg_installed/x86-windows/share/sentry/sentry_crashpad-targets.cmake:105 (set_target_properties):
1> [CMake]   The link interface of target "sentry_crashpad::zlib" contains:
1> [CMake] 
1> [CMake]     ZLIB::ZLIB
```
This problem can be solved by adding additional project dependency for zlib in vcpkg.
Related issue:https://github.com/getsentry/sentry-native/issues/596 
https://github.com/microsoft/vcpkg/issues/39114

### CODE REVIEW

File: sentry-config.cmake.in [hunk 1]
```diff
@@ -7,7 +7,7 @@ set(SENTRY_LINK_PTHREAD @SENTRY_LINK_PTHREAD@)
 
 if(SENTRY_BACKEND STREQUAL "crashpad" AND NOT SENTRY_BUILD_SHARED_LIBS)
 	include("${CMAKE_CURRENT_LIST_DIR}/sentry_crashpad-targets.cmake")
-	if(NOT MSVC AND NOT SENTRY_BUILD_SHARED_LIBS)
+	if(NOT SENTRY_BUILD_SHARED_LIBS)
```
Comment (supervacuus) [inline sentry-config.cmake.in:10]:
```suggestion
        set(CRASHPAD_ZLIB_SYSTEM @CRASHPAD_ZLIB_SYSTEM@)
	if(CRASHPAD_ZLIB_SYSTEM AND NOT SENTRY_BUILD_SHARED_LIBS)
```

**Review by supervacuus — CHANGES_REQUESTED** [review]
Thanks, @JonLiu1993. The reason for excluding `MSVC` here is because, by default, `crashpad` uses a vendored `zlib` when building on `MSVC`:

https://github.com/getsentry/crashpad/blob/202c1c21939f6b1d3c72ebeda566b4164a2fd245/CMakeLists.txt#L17-L26

https://github.com/getsentry/crashpad/blob/eb9a5e2d9da8fdfb1e8365fdc611a3916a82c275/third_party/zlib/CMakeLists.txt#L1-L38

Adding a system dependency to the consuming project when the build uses a vendored version fixes the wrong problem.

How about you check for `CRASHPAD_ZLIB_SYSTEM` instead? Would that fix the issue? The initial committer probably wanted to exclude that case without worrying about propagating the value from the `crashpad` project to `sentry-native`. If you wish to check on  `CRASHPAD_ZLIB_SYSTEM` in the exported config, you'd also have to define the option in `sentry-native`.

**Comment by dg0yt** [comment]
Outdated, invalid.

**Comment by supervacuus** [comment]
Hi @dg0yt and @JonLiu1993, can you minimally explain why this PR is no longer valid? Or provide a link to a `vcpkg` PR/issue that resolves the mentioned problem differently?

**Comment by JonLiu1993** [comment]
@supervacuus, VCPKG has updated sentry-native to 0.7.6 a few days ago. I thought this PR had been merged into it. I checked and found that it was not. I reopened this PR. Sorry for the confusion. 
https://github.com/microsoft/vcpkg/pull/39255/files

**Comment by dg0yt** [comment]
> Hi @dg0yt and @JonLiu1993, can you minimally explain why this PR is no longer valid?

It incorrectly reverts `find_package`.
It incorrectly relects the conditions.

**Comment by dg0yt** [comment]
Cf. #1011 for all recent comments transformed into a PR.

**Comment by supervacuus** [comment]
> It incorrectly reverts `find_package`.
> It incorrectly reflects the conditions.

True, I saw that, but that could have been fixed here.

> Cf. https://github.com/getsentry/sentry-native/pull/1011 for all recent comments transformed into a PR.

This includes the proposed feedback from here, so can we close this?

**Pull request closed without merging**